discovery-aws-api/src/main/scala/akka/discovery/awsapi/ecs/EcsSimpleServiceDiscovery.scala

```diff
@@ -0,0 +1,113 @@
+    import system.dispatcher
+    Future.firstCompletedOf(
+      Seq(
+        after(resolveTimeout, using = system.scheduler)(
+          Future.failed(new TimeoutException("Future timed out!"))
+        ),
+        Future {
+          Resolved(
+            serviceName = name,
+            addresses = for {
+              task <- resolveTasks(ecsClient, name)
+              container <- task.getContainers.asScala
+              networkInterface <- container.getNetworkInterfaces.asScala
+            } yield
+              ResolvedTarget(
+                host = networkInterface.getPrivateIpv4Address,
+                port = None
+              )
+          )
+        }
+      )
+    )
+  }
+}
+
+object EcsSimpleServiceDiscovery {
+
+  // InetAddress.getLocalHost.getHostAddress throws an exception when running
+  // in awsvpc mode because the container name cannot be resolved.
+  // ECS provides a metadata file
+  // (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/container-metadata.html)
+  // that we ought to be able to use instead to find our IP address, but the
+  // metadata file does not get set when running on Fargate. So this is our
+  // only option for determining what the canonical Akka and akka-management
+  // hostname values should be set to.
+  def getContainerAddress: Either[String, InetAddress] =
+    NetworkInterface.getNetworkInterfaces.asScala
+      .flatMap(_.getInetAddresses.asScala)
+      .filterNot(_.isLoopbackAddress)
+      .filter(_.isSiteLocalAddress)
+      .to[Seq] match {
+      case Seq(value) =>
+        Right(value)
+
+      case other =>
+        Left(s"Exactly one private address must be configured (found: $other).")
+    }
+
+  private def resolveTasks(ecsClient: AmazonECS, serviceName: String): Seq[Task] =
+    describeTasks(ecsClient, listTaskArns(ecsClient, serviceName))
+
+  @tailrec private[this] def listTaskArns(ecsClient: AmazonECS,
+                                          serviceName: String,
+                                          pageTaken: Option[String] = None,
+                                          accumulator: Seq[String] = Seq.empty): Seq[String] = {
+    val listTasksResult = ecsClient.listTasks(
+      new ListTasksRequest()
+        .withServiceName(serviceName)
+        .withNextToken(pageTaken.orNull)
+        .withDesiredStatus(DesiredStatus.RUNNING)
+    )
+    val accumulatedTasksArns = accumulator ++ listTasksResult.getTaskArns.asScala
+    listTasksResult.getNextToken match {
+      case null =>
+        accumulatedTasksArns
+
+      case nextPageToken =>
+        listTaskArns(
+          ecsClient,
+          serviceName,
+          Some(nextPageToken),
+          accumulatedTasksArns
+        )
+    }
+  }
+
+  private[this] def describeTasks(ecsClient: AmazonECS, taskArns: Seq[String]): Seq[Task] =
+    for {
+      // Each DescribeTasksRequest can contain at most 100 task ARNs.
+      group <- taskArns.grouped(100).to[Seq]
```

Comment:
should this be configurable? hmm

Reply:
I can't see an obvious reason for it to be.